lib/puppet/pops/issues.rb

```diff
@@ -347,20 +347,20 @@ def self.hard_issue(issue_code, *args, &block)
   # E.g. an arithmetic expression where a hostname is expected.
   #
   ILLEGAL_EXPRESSION = hard_issue :ILLEGAL_EXPRESSION, :feature, :container do
-    _("Illegal expression. %{expression} is unacceptable as %{feature} in %{container}") % { expression: label.a_an_uc(semantic), feature: feature, container: label.a_an(container) }
+    _("Illegal expression: %{expression}. It is unacceptable as %{feature} in an expression of type %{container}.") % { expression: label(semantic), feature: feature, container: label(container) }
```

Comment:
I think it's clearer without the sentence break. Suggest removing ". It"

Reply:
updated in latest commit.